src/contexts/wishlist-context.js

```diff
@@ -0,0 +1,85 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import { useNavigate } from "react-router-dom";
+import {useAuth} from "./auth-context"
+
+import {wishlistReducer} from "../reducer/wishlistReducer"
+const wishlistContext = createContext()
+
+const WishlistProvider =({children})=>{
+    const navigate = useNavigate();
+    const { isLogged } = useAuth()
+    const encodedToken = localStorage.getItem("encodedToken")
+    const initialState = {
+        wishItems:[],
+        wishCount:0
+    }
+
+    const [state, dispatch] = useReducer(wishlistReducer, initialState)
+    async function addToWishlist(product){
```

Comment:
can we shift this api part in separate file like main API OBJ
wishlistMethods = {
     addtoWishlist(){
          return axios.post(productdetails)
          // also you can add those checks()201 and 200 for response and return the final result / error 
     }
}

and call this object method here 

wishlistMethods.addToWishlist(details)

making it more cleaner

it just an opinion , you code is functionally good and easy to understand , so you can do this refactoring aftewards

Reply:
Yeah, I thought about this. I will do this

Thank You